src/openrct2/Imaging.cpp

```diff
@@ -97,6 +102,16 @@ namespace Imaging
                     }
                 }
             }
+            else if (bitDepth == 8 && !expand)
+            {
+                // 8-bit paletted
+                Guard::Assert(rowBytes == pngWidth, GUARD_LINE);
```

Comment:
Isn't there any better way to verify if image is paletted? libpng's docs state:

> 
    channels       - number of channels of info for the
                     color type (valid values are 1 (GRAY,
                     PALETTE), 2 (GRAY_ALPHA), 3 (RGB),
                     4 (RGB_ALPHA or RGB + filler byte))

So you could be loading a grayscale image here. Keep in mind the palette can be smaller than 256 entries.

A recent PR caused discussion about how to handle errors on invalid files (https://github.com/OpenRCT2/OpenRCT2/pull/6325/files#r141807768). To keep it more in line, change it to throw rather than assert.

Reply:
Yes, this is a bit of an awkward bit of code. It actually never checks for palettes, it just checks for 8bpp.

Might it work to say that when `expand` is set the function should always return a 4-channel (32bpp) image, and otherwise simply returns the image as bytes with how-ever many bits per pixel. We then add `bitDepth` as an out parameter.

This way `expand` just becomes a 'guarantee true colour' option, and the lack of it means the function requesting the bits is responsible for any interpretation. Of course the function could also pass back the PNG's palette, but I am not sure that is needed. The only two uses now are to load full-colour images (from what-ever source) and to load 8bpp images which are interpreted as paletted by RCT's palette.

Reply:
`bitDepth` sounds like a good-enough solution to me.